notes/lecture7.tex

```diff
@@ -0,0 +1,137 @@
+\section{Lecture 7: Nesterov’s accelerated gradient descent}
+
+In this lecture, we present Nesterov's accelerated gradient descent for smooth convex functions. We will derive an algorithm that obtains a runtime of $\cO \left(\frac{\beta}{t^2}\right)$ for $\beta$-smooth functions. For functions which are $\alpha$-strongly convex, we will achieve a rate of $\cO \left(\exp\left( -\sqrt{\frac{\beta}{\alpha}} t \right) \right)$. The algorithm proceeds iteratively as follows. 
+\begin{align*}
+x_0 &= y_0 = z_0, \\
+y_k &= x_k - \frac{1}{\beta} \nabla f(x_k), \\
+z_k &= z_{k -1} - \eta \|\nabla f(x_k)\|, \\
+x_{k+1} &= \tau z_k + (1 - \tau) y_k.
+\end{align*}
+The definitions are constructed so that we can apply bounds related to smooth and convex functions. We will simplify our proof of the runtime of the algorithm by ``restarting'' the procedure in the lemma several times; namely, after the error is reduced from $d$ to $\frac{d}{2}$, we run our algorithm again with a new $x_0 = x_T$
+
+\begin{lemma}
+Suppose $\|x_0 - x^* \| \leq R$, $f$ attains its minimum at $x^*$ and is $\beta$-smooth. Also assume $f(x_0) -f(x^*) \leq d$.
+Put $\eta = \frac{R}{\sqrt{d\beta}}$, and $\tau$ s.t. $\frac{1-\tau}{\tau} = \eta \beta$.\\
+Then after $T = 4R\sqrt{\frac{\beta}{d}}$ steps, we have $$f(\bar{x})- f(x^*) \leq \frac{d}{2},$$ where $\bar{x} = \frac{1}{T} \sum_{k=0}^{T-1} x_k$.
+\end{lemma}
+% I think we formalize this argument later.
+% What do we get if we ``restart'' the lemma several times?
+% We get that after
+% $$T = O\left(R \sqrt{\frac{\beta}{\epsilon}} + R \sqrt{\frac{\beta}{2\epsilon}} +...+ R \sqrt{\frac{\beta}{d}} \right) = O\left(\sqrt{\frac{\beta}{\epsilon}}\right)$$
+% steps we get $\epsilon$-error.
+% This is equivalent to stating that the error $\epsilon$ is $O\left(\frac{R\beta}{T^2}\right)$.
+
+\begin{proof}
+By smoothness and convexity, we have
+\begin{equation}
+f(y_k) - f(x_k) \leq -\frac{1}{2 \beta} \|\nabla f(x_k) \|^2. \label{Nesterov-pf_smoothness_and_convexity}
+\end{equation}
+By convexity, we have
+\begin{equation*}
+\forall u: \ \eta  \langle \nabla f(x_{k+1}), z_k - u \rangle \leq \frac{\eta^2}{2} \|\nabla f(x_{k+1})\|^2 + \|z_k - u \|^2 - \|z_{k+1} - u \|^2.
```

Comment:
As discussed on Piazza, the squared norm terms should have a factor `1/2` in front of each of them. Once you write it like that it holds with equality.